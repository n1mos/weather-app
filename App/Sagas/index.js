import { takeLatest, all, fork } from 'redux-saga/effects'
import { WeatherTypes } from 'App/Stores/Weather/Actions'
import { StartupTypes } from 'App/Stores/Startup/Actions'
import { fetchWeather, watchWeather } from './WeatherSaga'
import { startup } from './StartupSaga'

export default function* root() {
  yield all([
    takeLatest(StartupTypes.STARTUP, startup),
    takeLatest(WeatherTypes.FETCH_WEATHER, fetchWeather),
    fork(watchWeather)
  ])
}
