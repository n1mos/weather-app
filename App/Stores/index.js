import { combineReducers } from 'redux'
import configureStore from './CreateStore'
import rootSaga from 'App/Sagas'
import { reducer as WeatherReducer } from './Weather/Reducers'

export default () => {
  const rootReducer = combineReducers({
    weather: WeatherReducer,
  })

  return configureStore(rootReducer, rootSaga)
}
