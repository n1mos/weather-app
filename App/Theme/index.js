import Colors from './Colors'
import Fonts from './Fonts'
import Images from './Images'
import Dimensions from './Dimensions'

export { Colors, Fonts, Images, Dimensions }
