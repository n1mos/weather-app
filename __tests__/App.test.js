import React from 'react'
import { mount } from 'enzyme'
import App from '../App/App'

test('App renders correctly', () => {
  const wrapper = mount(<App />)

  expect(wrapper).toMatchSnapshot()
})