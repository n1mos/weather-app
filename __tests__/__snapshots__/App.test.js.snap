// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<App>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <PersistGate
      loading={null}
      persistor={
        Object {
          "dispatch": [Function],
          "flush": [Function],
          "getState": [Function],
          "pause": [Function],
          "persist": [Function],
          "purge": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
    />
  </Provider>
</App>
`;
